### PR TITLE
Added new feature: restore incrementally

### PR DESCRIPTION
**Pull request by shuo-wu**

### CODE REVIEW

File: integration/data/utils.py [hunk 1]
```diff
@@ -38,20 +38,20 @@ def create_backup(backup_target, snap):
     return backup
 
 
-def rm_backups(*backups):
+def rm_backups(url, *backups):
```
Comment (yasker) [inline integration/data/utils.py]: We can change *backups to an array and put `url` after it. It looks weird to have the empty inserted into every function.

File: integration/data/cmd.py [hunk 2]
```diff
@@ -17,69 +21,83 @@ def _bin():
...
+def info(url=CONTROLLER):
+    cmd = [_bin(), '--url', url]
+    cmd = cmd + ['--debug', 'info']
     return json.loads(subprocess.check_output(cmd))
 
 
-def snapshot_create():
-    cmd = [_bin(), '--debug', 'snapshot', 'create']
+def snapshot_create(url=CONTROLLER):
+    cmd = [_bin(), '--url', url]
```
Comment (yasker) [inline integration/data/cmd.py]:
```
def longhorn(url):
    return [_bin(), '--url', url]

def info(url=CONTROLLER):
    cmd = longhorn(url) + ['--debug', 'info']
    return json.loads(subprocess.check_output(cmd))
```

File: integration/data/common.py [hunk 3]
```diff
@@ -31,9 +31,15 @@
 UPGRADE_REPLICA2 = 'tcp://localhost:9515'
 UPGRADE_REPLICA2_SCHEMA = 'http://localhost:9515/v1/schemas'
 
-CONTROLLER = "http://localhost:9501"
+STANDBY_REPLICA1 = 'tcp://localhost:9522'
+STANDBY_REPLICA1_SCHEMA = 'http://localhost:9522/v1/schemas'
+STANDBY_REPLICA2 = 'tcp://localhost:9525'
+STANDBY_REPLICA2_SCHEMA = 'http://localhost:9525/v1/schemas'
+
+STANDBY_REPLICA1_PATH = '/tmp/fix1/'
```
Comment (yasker) [inline integration/data/common.py]: `fix1` is weird. How about `/tmp/standby_vol1`?

And this should be in the next commit?

File: integration/data/cmd.py [hunk 4]
```diff
@@ -108,3 +108,10 @@ def restore_to_file(backup_url, backing_file='', output_file='', format='',
         cmd.append('--output-format')
         cmd.append(format)
     return subprocess.check_output(cmd)
+
+
+def restore_incre(backup_url, last_restored, url=CONTROLLER):
```
Comment (yasker) [inline integration/data/cmd.py]: `restore_inc` is better.

File: integration/data/cmd.py [hunk 5]
```diff
@@ -108,3 +108,10 @@ def restore_to_file(backup_url, backing_file='', output_file='', format='',
...
+    cmd = [_bin(), '--url', url]
```
Comment (yasker) [inline integration/data/cmd.py]: Same as before.

File: integration/data/common.py (hunk 3 above)
Comment (shuo-wu) [reply integration/data/common.py]: vendor file commit is the 1st first. met github bug again...
In engine command, the short name of flag `--incrementally` is `-i`. Is that OK?

Comment (yasker) [reply integration/data/common.py]: I think you made it `-I`, but either is OK.

Comment (yasker) [reply integration/data/common.py]: Double check.

**Pull request merged**